modules/images/webp-uploads/load.php

```diff
@@ -603,3 +603,78 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 }
 
 add_filter( 'the_content', 'webp_uploads_update_image_references', 10 );
+
+/**
+ * Returns the attachment sources array ordered by filesize.
+ *
+ * @since n.e.xt
+ * @private
+ *
+ * @param int    $attachment_id The attachment ID.
+ * @param string $size          The attachment size.
+ *
+ * @return array The attachment sources array.
+ */
+function webp_uploads_get_attachment_sources( $attachment_id, $size = 'thumbnail' ) {
+	// Check for the sources attribute in attachment metadata.
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+
+	// Return full image size sources.
+	if ( 'full' === $size && ! empty( $metadata['sources'] ) ) {
+		return $metadata['sources'];
+	}
+
+	// Return the resized image sources.
+	if ( isset( $metadata['sizes'][ $size ]['sources'] ) && ! empty( $metadata['sizes'][ $size ]['sources'] ) ) {
+		return $metadata['sizes'][ $size ]['sources'];
+	}
+
+	// Return an empty array if no sources found.
+	return [];
```

Comment:
We are using the long syntax array for arrays due to WP styleguide.

```suggestion
	return array();
```